common/src/node.rs

```diff
@@ -77,6 +96,39 @@ impl Graph {
         Some((chunk, coords))
     }
 
+    /// Populates a chunk with the given voxel data and ensures that margins are correctly removed
+    /// if necessary.
+    pub fn populate_chunk(&mut self, chunk: ChunkId, mut new_data: VoxelData, modified: bool) {
+        // New chunks generated by world generation should not be solid if they are adjacent to modified chunks.
+        if !modified && new_data.is_solid() {
+            for coord_axis in 0..3 {
+                for coord_direction in [1, -1] {
+                    if let Some(chunk_id) =
+                        self.get_chunk_neighbor(chunk, coord_axis, coord_direction)
+                    {
+                        if let Chunk::Populated { modified: true, .. } = self[chunk_id] {
+                            // Remove margins if next to a modified chunk
```

Comment:
It does after one of the edits I made to it. It is definitely rather confusing, so it would make more sense for me to separate the logic of removing margins from the logic of `data_mut`.